tsl/src/bgw_policy/job.c

```diff
@@ -484,6 +575,61 @@ policy_compression_read_and_validate_config(Jsonb *config, PolicyCompressionData
 	}
 }
 
+void
+policy_recompression_read_and_validate_config(Jsonb *config, PolicyCompressionData *policy_data)
+{
+	Oid table_relid = ts_hypertable_id_to_relid(policy_compression_get_hypertable_id(config));
+	Cache *hcache;
+	Hypertable *hypertable =
+		ts_hypertable_cache_get_cache_and_entry(table_relid, CACHE_FLAG_NONE, &hcache);
+	if (policy_data)
+	{
+		policy_data->hypertable = hypertable;
+		policy_data->hcache = hcache;
+	}
+}
+
+bool
+policy_recompression_execute(int32 job_id, Jsonb *config)
+{
+	int32 chunkid;
+	Dimension *dim;
+	PolicyCompressionData policy_data;
+
+	policy_recompression_read_and_validate_config(config, &policy_data);
+	dim = hyperspace_get_open_dimension(policy_data.hypertable->space, 0);
+	chunkid = get_chunk_to_recompress(dim, config);
+
+	if (chunkid == INVALID_CHUNK_ID)
+		elog(NOTICE,
+			 "no chunks for hypertable %s.%s that satisfy recompress chunk policy",
```

Comment:
```suggestion
			 "no chunks for hypertable \"%s.%s\" that satisfy recompress chunk policy",
```

Identifier names are usually double quoted in messages.

Reply:
fixed.